src/tools/wasm-shell.cpp

```diff
@@ -304,9 +304,9 @@ int main(int argc, const char* argv[]) {
         std::cerr << "BUILDING MODULE [line: " << curr.line << "]\n";
         Colors::normal(std::cerr);
         auto module = wasm::make_unique<Module>();
-        Name moduleName;
+        auto& moduleName = module->name;
         auto builder = wasm::make_unique<SExpressionWasmBuilder>(
-          *module, *root[i], IRProfile::Normal, &moduleName);
+          *module, *root[i], IRProfile::Normal);
```

Comment:
Is this fine?

Reply:
Ah, I got confused by the name being referenced before `SExpressionWasmBuilder` runs and populates it. Looking at this again, seems like it's safe to just copy the name when moving this line below the builder constructor.